tools/imas/imas_wrapper.py

```diff
@@ -0,0 +1,542 @@
+	
+	# get the indices fot the limits in psi_eq
+	# get the magnetic axis index
+	magnetic_axis_psi_index = np.where(eq_psi == max(eq_psi))[0][0]
+	eq_psi = eq_psi[magnetic_axis_psi_index:]
+	
+	#normalize eq_psi
+	eq_psi = eq_psi - eq_psi[0]	
+
+	# load the coreprof psi
+	if len(coreprof.profiles_1d[0].grid.psi) == 0:
+		logging.warning('The poloidal flux from core profiles IDS seems to be empty!\n')
+		raise Exception('Error in loading core profiles data. Please check the input data!')
+	else:
+		cp_psi = coreprof.profiles_1d[0].grid.psi
+		if log:
+			logging.info('The poloidal flux from core profiles IDS loaded successfully!\n')
+	
+	# interpolate the coreprof rho from the coreprof psi to the equilibrium psi
+	interp_radius = CubicSpline(-cp_psi, cp_radius)
+	radius = interp_radius(-eq_psi)
+	
+	# Check if the interpolation gave a correct answer
+	error = (radius[-1] - cp_radius[-1]) / radius[-1]
+	if abs(error) > 0.1:
+		raise Exception("The interpolation gave significant difference in the equilibrium and coreprofile radii. Cannot continue, please check the input parameters.")
+	
+	#load wall data with error handling
+	if wall.description_2d[0].vessel.unit[0].annular.resistivity == -9e+40:
+		logging.warning('The the wall resistivity from wall IDS seems to be empty!\n')
+		raise Exception('Error in loading wall data. Please check the input data!')
+	else:
+		resistivity = wall.description_2d[0].vessel.unit[0].annular.resistivity
+		if log:
+			logging.info('The major radius of the magnetic axis from core profiles IDS loaded successfully!\n')
+
+	if setUpDream:
+		
+		ds = DREAMSettings()
+		
+		ds.radialgrid.setCustomGridPoints(radius)
+
+		if wall_radius == -999:
+			raise Exception('If you want to set up a dream settings object, please specify the wall radius!')
+		else:
+			ds.radialgrid.setWallRadius(wall_radius)
+			
+		ds.radialgrid.setMajorRadius(R)
+		ds.radialgrid.setB0(abs(B0))
+		
+		for i in range(len(coreprof.profiles_1d[0].ion)):
+			ds.eqsys.n_i.addIon(ion_names[i], Z=ion_charges[i], iontype=Ions.IONS_DYNAMIC, Z0=ion_charges[i], n=ion_densities[i], r=cp_radius)
+			
+		ds.eqsys.T_cold.setInitialProfile(T_cold, radius=cp_radius)
+		ds.eqsys.T_cold.setType(Tcold.TYPE_SELFCONSISTENT)
+		
+		if e_field!=-999:
+			ds.eqsys.E_field.setType(ElectricField.TYPE_SELFCONSISTENT)
+			ds.eqsys.E_field.setInitialProfile(efield=e_field, radius=cp_radius)
+		
+		else:
+			inverse_wall_time = resistivity / (4*np.pi*1e-7*R*np.log(R/wall_radius))
+			if log:
+				logging.info('The iverse wall time is %s 1/s with the given wall radius.\n', inverse_wall_time)
+		
+			ds.eqsys.E_field.setType(ElectricField.TYPE_SELFCONSISTENT)
+			ds.eqsys.E_field.setBoundaryCondition(ElectricField.BC_TYPE_SELFCONSISTENT, inverse_wall_time=inverse_wall_time, R0=R)
+		
+		ds.eqsys.j_ohm.setInitialProfile(j_tor, radius=eq_radius, Ip0=Ip)
+		
+		return ds
+	
+	else:
+		print('return a list? hdf5?')
+		return 'TO DO'
```

Comment:
This seems like something production code shouldn't do? It seems you have been setting a bunch of variables so far? If so, I would suggest return a ``dict`` with those variables here.

Reply:
I agree and putting the into a `dict` seems the simplest solution. It is implemented, and the documentation is modified accordingly.